tests/run-make/rust-lld-compress-debug-sections/rmake.rs

```diff
@@ -0,0 +1,39 @@
+// Checks the `compress-debug-sections` option on rust-lld.
+
+//@ needs-rust-lld
+//@ only-linux
+//@ ignore-cross-compile
+
+// FIXME: This test isn't comprehensive and isn't covering all possible combinations.
+
+use run_make_support::{assert_contains, cmd, llvm_readobj, run_in_tmpdir, rustc};
```

Comment:
@khuey To elaborate a bit why it might be desirable to extend compiletest a bit here: Currently, this test doesn't guarantee we actually test that zstd and zlib compression works, as noted in this FIXME.

We could at least test both positive cases if the necessary case recognition was added to compiletest's [headers](https://rustc-dev-guide.rust-lang.org/tests/headers.html), so that we can implement what you currently do using a runtime `if` by instead using compiletest's [revisions](https://rustc-dev-guide.rust-lang.org/tests/compiletest.html#revisions).

Reply:
> so that we can implement what you currently do using a runtime `if` by instead using compiletest's [revisions](https://rustc-dev-guide.rust-lang.org/tests/compiletest.html#revisions).

Unfortunately, revisions are only supported by a subset of test suites, it is not supported in run-make tests. However, we *could* try to pass info from compiletest about whether zstd/zlib is available in the environment (or I'm not sure if it's possible, prevent a sub test case from finding zstd/zlib in the environment).

Reply:
Ahh, I keep forgetting the arbitrary limitations compiletest has, my mistake.